### PR TITLE
Use expected_width parameter in flattening/boolbv_if to simplify error handling

### PR DESCRIPTION
**Pull request by hannes-steffenhagen-diffblue**
- [ ] Each commit message has a non-empty body, explaining why the change was made.
- [x] My contribution is formatted in line with CODING_STANDARD.md.
- [x] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [ ] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- [x] My commit message includes data points confirming performance improvements (if claimed) (**not applicable**).
- [x] My PR is restricted to a single feature or bugfix.
- [x] White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 3779e522d85b36e37e9600eb0bb9eab438bfd938).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/85890465

File: src/solvers/flattening/boolbv_if.cpp [hunk 1]
```diff
@@ -18,11 +18,11 @@ bvt boolbvt::convert_if(const if_exprt &expr)
 
   literalt cond=convert(expr.cond());
 
-  const bvt &op1_bv=convert_bv(expr.true_case());
-  const bvt &op2_bv=convert_bv(expr.false_case());
+  const bvt &true_case_bv = convert_bv(expr.true_case());
+  const bvt &false_case_bv = convert_bv(expr.false_case());
 
-  if(op1_bv.size()!=width || op2_bv.size()!=width)
+  if(true_case_bv.size() != width || false_case_bv.size() != width)
```
Comment (tautschnig) [inline src/solvers/flattening/boolbv_if.cpp]: I think `width` should just be passed as second argument to `convert_bv` thanks to your earlier work?!

**Review by tautschnig — CHANGES_REQUESTED** [review]

File: src/solvers/flattening/boolbv_if.cpp (hunk 1 above)
Comment (NlightNFotis) [reply src/solvers/flattening/boolbv_if.cpp]: @hannes-steffenhagen-diffblue `convert_bv` in `boolbv.cpp` was changed recently to take an optional size. https://github.com/diffblue/cbmc/blob/develop/src/solvers/flattening/boolbv.cpp#L114 That makes the invariant present here as well unnecessary.

**Review by NlightNFotis — CHANGES_REQUESTED** [review]
The data invariant is not needed, just pass the width as an extra argument to `convert_bv`. The invariant is being enforced inside `convert_bv` now.

**Comment by hannes-steffenhagen-diffblue** [comment]
Apologies, the changes to `convert_bv` hadn't been made when I committed these changes. I'll revise them.

**Comment by hannes-steffenhagen-diffblue** [comment]
@NlightNFotis @tautschnig I believe I've addresses your comments, can you rereview?

**Review by tautschnig — APPROVED** [review]

**Review by NlightNFotis — APPROVED** [review]
Looks good to me.

**Review by allredj — COMMENTED** [review]
Passed Diffblue compatibility checks (cbmc commit: 2ab7ac72749b04ca8bad3fcb3fc3dc950bb74542).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/86025732

**Pull request merged**